### PR TITLE
[3 NEW MODULES] Added Glide, CreativeFly and GhostMode :+1:

### PR DESCRIPTION
**Pull request by PiotrekUnityDeveloper**
A new glide module!
module settings (vertical mode is for development testing and for fun):
image
example of glide module usage (sorry for lag and bad quality, i hav potato pc):

https://user-images.githubusercontent.com/85606121/160300923-a8a79c7b-9774-492c-a9ee-1ed17cfa1928.mp4

==================================================================
second module! CreativeFly!
image
double press space to toggle flying!
example usage of the module:

https://user-images.githubusercontent.com/85606121/160300969-f7a2da9c-f92f-493c-bac8-f6dfa2a0e17f.mp4

============================================================================
third module: GhostMode (from meteorrejects)
Gives player an ability to walk after hes dead
image

video showcase:

https://user-images.githubusercontent.com/85606121/160302510-bb5aac55-4fa4-4ad3-a131-625ed5f83c6a.mp4

==================
thanks :D

### CODE REVIEW

**Comment by PiotrekUnityDeveloper** [comment]
and not, this is not skidded

File: src/main/java/mathax/client/systems/modules/player/GhostMode.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	
+	 private final Setting<Boolean> fullFood = sgGeneral.add(new BoolSetting.Builder()
+		        .name("full-food")
+		        .description("Sets the food level client-side to max so u can sprint.")
+		        .defaultValue(true)
+		        .build()
+		    );
+	 
+    public GhostMode() {
+        super(Categories.Player, Items.SKELETON_SKULL, "ghost mode", "You will be able to move after death");
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/player/GhostMode.java:30]: Change it to:

super(Categories.Player, Items.SKELETON_SKULL, "ghost", "Allows you to move after death.");

File: src/main/java/mathax/client/systems/modules/player/GhostMode.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        
+    }
+    
+    public void onDeactivate() {
+    	super.onDeactivate();
+        active = false;
+        warning("You are no longer in a ghost mode!");
+        if (mc.player != null && mc.player.networkHandler != null) {
+            mc.player.requestRespawn();
+            info("Respawn request has been sent to the server.");
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/player/GhostMode.java:60]: Remove this, we already told the player they are respawning.

File: src/main/java/mathax/client/systems/modules/player/GhostMode.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+    
+    @EventHandler
+    private void onOpenScreen(OpenScreenEvent event) {
+        if (event.screen instanceof DeathScreen) {
+            event.cancel();
+            if (!active) {
+                active = true;
+                info("You are now in a ghost mode. ");
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/player/GhostMode.java:70]: Changed to:

warning("You are now in ghost mode.");

File: src/main/java/mathax/client/systems/modules/player/GhostMode.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    }
+    
+    @EventHandler
+    private void onGameJoin(GameJoinedEvent event) {
+        active = false;
+    }
+    
+    public void onActivate() {
+        
+    }
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/player/GhostMode.java:52]: Why is this here?

File: src/main/java/mathax/client/systems/modules/player/GhostMode.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    private boolean active = false;
+
+    @EventHandler
+    private void onTick(final TickEvent.Pre event) {
+    	
+    	if (!active) return;
+        if (mc.player.getHealth() < 1f) mc.player.setHealth(20f);
+        if (fullFood.get() && mc.player.getHungerManager().getFoodLevel() < 20) {
+            mc.player.getHungerManager().setFoodLevel(20);
+        }
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/player/GhostMode.java:42]: Looks better:

if (fullFood.get() && mc.player.getHungerManager().getFoodLevel() < 20) mc.player.getHungerManager().setFoodLevel(20);

File: src/main/java/mathax/client/systems/modules/movement/Glide.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	        .name("strafing speed")
+	        .description("How fast to strafe?")
+	        .defaultValue(0.2)
+	        .min(0.001)
+	        .sliderRange(0.050, 1.0)
+	        .build()
+	    );
+	
+    public Glide() {
+        super(Categories.Movement, Items.FEATHER, "glide", "makes you glide slowly when falling");
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/movement/Glide.java:59]: Change to:

super(Categories.Movement, Items.FEATHER, "glide", "Makes you glide slowly while falling.");

File: src/main/java/mathax/client/systems/modules/movement/Glide.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    @EventHandler
+    private void onTick(final TickEvent.Post event) {
+    	
+    	if(defstrafe.get() == false)
+        {
+        	mc.player.airStrafingSpeed = strafespeed.get().floatValue();
+        }
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/movement/Glide.java:68]: I would prefer:

if (defstrafe.get() == false) mc.player.airStrafingSpeed = strafespeed.get().floatValue();

File: src/main/java/mathax/client/systems/modules/movement/CreativeFly.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    
+    @Override
+    public void onActivate() {
+    	mc.player.getAbilities().allowFlying = true;
+    	
+    	if(instafly.get() == true)
+    	{
+    		mc.player.getAbilities().flying = true;
+    	}
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/movement/CreativeFly.java:48]: I would prefer:

if (instafly.get() == true) mc.player.getAbilities().flying = true;

File: src/main/java/mathax/client/systems/modules/movement/CreativeFly.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+	
+	private final Setting<Boolean> tickenabler = sgGeneral.add(new BoolSetting.Builder()
+	        .name("update every tick")
+	        .description("updates your flying ability every tick, enable only if flying doesnt work without this turned on")
+	        .defaultValue(true)
+	        .build()
+	    );
+	
+    public CreativeFly() {
+        super(Categories.Movement, Items.CHAINMAIL_BOOTS, "creative fly", "makes you can fly like in creative mode");
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/movement/CreativeFly.java:38]: Change to:

super(Categories.Movement, Items.CHAINMAIL_BOOTS, "creative-fly", "Allows you to fly like in creative mode.");

File: src/main/java/mathax/client/systems/modules/Modules.java [hunk 10]
```diff
@@ -416,6 +416,7 @@ private void initPlayer() {
         add(new EXPThrower());
         add(new FastUse());
         add(new GhostHand());
+        add(new GhostMode()); //added by Piotrek4 (https://github.com/AntiCope/meteor-rejects/blob/master/src/main/java/anticope/rejects/modules/GhostMode.java)
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/Modules.java:419]: Add the credits to the module itself, like in ChatEncryption, Anchor Aura etc.

File: src/main/java/mathax/client/systems/modules/Modules.java [hunk 11]
```diff
@@ -443,12 +444,14 @@ private void initMovement() {
         add(new BoatPhase());
         add(new ChorusExploit());
         add(new ClickTP());
+        add(new CreativeFly()); //added by Piotrek4
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/Modules.java:447]: Add the credits to the module itself, like in ChatEncryption, Anchor Aura etc.

File: src/main/java/mathax/client/systems/modules/Modules.java [hunk 12]
```diff
@@ -443,12 +444,14 @@ private void initMovement() {
...
         add(new ElytraBoost());
         add(new ElytraFly());
         add(new EntityControl());
         add(new EntitySpeed());
         add(new FastClimb());
         add(new Flight());
+        add(new Glide()); //added by Piotrek4
```
Comment (Matejko06) [inline src/main/java/mathax/client/systems/modules/Modules.java:454]: Add the credits to the module itself, like in ChatEncryption, Anchor Aura etc.

**Review by Matejko06 — CHANGES_REQUESTED** [review]
Replace spaces in module and module setting names to -s, update descriptions to start with capital letter, end with a dot and make more sense. More details can be found in comments on this pr,

**Pull request merged**